src/Autofac/Core/Lifetime/LifetimeScope.cs

```diff
@@ -316,6 +316,41 @@ public object GetOrCreateAndShare(Guid id, Func<object> creator)
             return result;
         }
 
+        /// <summary>
+        /// Try to create an instance with a GUID key.
+        /// </summary>
+        /// <param name="id">Key.</param>
+        /// <param name="creator">Creation function.</param>
+        /// <returns>An instance.</returns>
+        public object Create(Guid id, Func<object> creator)
```

Comment:
It appears the `GetOrCreateAndShare` method was broken up into two pieces. Is there anything else calling the single unified version of the method? If not, and if the interface is already breaking, perhaps removing the old one is interesting so we're not maintaining it.

Reply:
Hi @tillig  I have just reviewed the code, the `GetOrCreateAndShare` method is not used anymore but maybe someone using Autofac library is calling it. In my opinion the `Get` method must be added (it does only "get" responsibility so I don't think it's a breaking change or has side-effect to anything else), the `Create` method maybe replaced by re-using `GetOrCreateAndShare` method if you want ;)